test/ut3_user/api/test_ut_run.pkb

```diff
@@ -787,5 +787,214 @@ Failures:%
+    l_results :=  ut3_tester_helper.main_helper.get_dbms_output_as_clob();
+    --Assert
+    ut.expect( l_results ).not_to_be_like( '%test_package_1.%executed%' );
+    ut.expect( l_results ).to_be_like( '%test_package_1%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2.%executed%' );
+    ut.expect( l_results ).not_to_be_like( '%test_package_3.%executed%' );  
+  end;
+ 
+  procedure two_test_run_by_one_tag is
+    l_results clob;
+  begin
+    ut3_tester_helper.run_helper.run(a_tags => 'test2');
+    l_results :=  ut3_tester_helper.main_helper.get_dbms_output_as_clob();
+    --Assert
+    ut.expect( l_results ).not_to_be_like( '%test_package_1.%executed%' );
+    ut.expect( l_results ).to_be_like( '%test_package_1%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2.%executed%' );
+    ut.expect( l_results ).not_to_be_like( '%test_package_3.%executed%' );  
+  end;
+  
+  procedure all_suites_run_by_one_tag is
+    l_results clob;
+  begin
+    ut3_tester_helper.run_helper.run(a_tags => 'helper');
+    l_results :=  ut3_tester_helper.main_helper.get_dbms_output_as_clob();
+    --Assert
+    ut.expect( l_results ).to_be_like( '%test_package_1%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2%' );
+    ut.expect( l_results ).to_be_like( '%test_package_3%' ); 
+  end; 
+  
+  procedure two_test_run_by_two_tags is
+    l_results clob;
+  begin
+    ut3_tester_helper.run_helper.run(a_tags => 'subtest1,subtest2');
+    l_results :=  ut3_tester_helper.main_helper.get_dbms_output_as_clob();
+    --Assert
+    ut.expect( l_results ).to_be_like( '%test_package_1%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2%' );
+    ut.expect( l_results ).not_to_be_like( '%test_package_1.test2%' ); 
+    ut.expect( l_results ).not_to_be_like( '%test_package_2.test2%' ); 
+    ut.expect( l_results ).not_to_be_like( '%test_package_3%' ); 
+    ut.expect( l_results ).not_to_be_like( '%test_package_3%' );  
+  end;
+  
+  procedure suite_with_children_tag  is
+    l_results clob;
+  begin
+    ut3_tester_helper.run_helper.run(a_tags => 'suite1');
+    l_results :=  ut3_tester_helper.main_helper.get_dbms_output_as_clob();
+    --Assert
+    ut.expect( l_results ).to_be_like( '%test_package_1%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2%' );
+    ut.expect( l_results ).not_to_be_like( '%test_package_3%' );  
+  end;
+  
+  procedure suite_with_tag_parent is
+    l_results clob;
+  begin
+    ut3_tester_helper.run_helper.run(a_tags => 'suite2');
+    l_results :=  ut3_tester_helper.main_helper.get_dbms_output_as_clob();
+    --Assert
+    ut.expect( l_results ).to_be_like( '%test_package_1%' );
+    ut.expect( l_results ).to_be_like( '%test_package_2%' );
+    ut.expect( l_results ).not_to_be_like( '%test_package_3%' );  
+  end;
+  
+  procedure test_nonexists_tag is
```

Comment:
This is unexpected behavior.
We should change this.
When running all tests and filtering by tags, and no tagged items found, the run returns 0 (zero) tests executed.